File: src/screens/onboarding/NameOnboarding.js
import { View, Text, StyleSheet, TextInput } from 'react-native'
import React, { useState } from 'react'
import { SafeAreaView } from 'react-native-safe-area-context'
import colors from '../../utilities/colors'
import ArrowButton from '../../components/onboarding/ArrowButton'
import BackButton from '../../components/onboarding/BackButton'

export default function NameOnboarding({ onNext, onBack, setUserData }) {

    const [text, onChangeText] = useState('');
    const onPress = () => {
        setUserData(prev => prev.name = text)
        onNext()
    }

    return (
        <SafeAreaView style={style.container}>
            <View>
                <BackButton onPress={onBack} />
            </View>
            <View style={style.content}>
                <Text style={style.title}>What is your name? </Text>
                <Text style={style.subtitle}>Please enter your name so that we can address you properly.</Text>
                <TextInput
                    style={style.input}
                    onChangeText={onChangeText}
                    placeholder={'Name'}
                    placeholderTextColor={'gray'}
                    value={text} />
            </View>
            <View style={style.console}>
                {text.trim() !== '' && <ArrowButton onPress={onPress} />}

            </View>

        </SafeAreaView>
    )
}
const style = StyleSheet.create({
    container: {
        backgroundColor: colors.black,
        flex: 1,
        paddingTop: 10,
        paddingHorizontal: 20,
    }, content: {
        flex: 1,
    },
    title: {
        color: colors.white,
        fontSize: 30,
        fontWeight: 'bold'
    }, subtitle: {
        fontSize: 15,
        color: colors.white,
        fontWeight: 'bold',
        marginTop: 10
    }, input: {
        backgroundColor: colors.white,
        height: 50,
        width: '100%',
        marginTop: 10,
        borderRadius: 10,
        paddingHorizontal: 10,
        fontWeight: 'bold'


    }, console: {
        alignItems: 'flex-end',
        marginBottom: 20
    }
})

File: lib/screens/onboarding/name_onboarding.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:zihnai/generated/l10n.dart';
import 'package:zihnai/providers/user_provider.dart';
import 'package:zihnai/ultils/color.dart';

class NameOnboarding extends StatefulWidget {
  const NameOnboarding({super.key, required this.onNext, required this.onBack});
  final void Function() onNext;
  final void Function() onBack;

  @override
  NameOnboardingState createState() => NameOnboardingState(); // Removed the leading underscore
}

class NameOnboardingState extends State<NameOnboarding> {
  TextEditingController textController = TextEditingController();

  @override
  void initState() {
    super.initState();
    textController.text = context.read<UserProvider>().name;
  }

  @override
  void dispose() {
    textController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    void onPress() {
      widget.onNext();
    }

    return SafeArea(
      child: Scaffold(
        backgroundColor: CustomColors.dark,
        body: Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 10.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Back Button
              BackButton(onPressed: widget.onBack, color: Colors.white),

              // Content Section
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      S.of(context).onboardingNameTitle,
                      style: TextStyle(
                        fontSize: 30,
                        fontWeight: FontWeight.bold,
                        color: Colors.white,
                      ),
                    ),
                    SizedBox(height: 10),
                    Text(
                      S.of(context).onboardingNameSubtitle,
                      style: TextStyle(fontSize: 15, color: Colors.grey),
                    ),
                    SizedBox(height: 20),
                    TextField(
                      controller: textController,
                      onChanged: (value) {
                        context.read<UserProvider>().setName(value);
                      },
                      style: TextStyle(fontWeight: FontWeight.bold),
                      decoration: InputDecoration(
                        hintText: 'Name',
                        hintStyle: TextStyle(color: Colors.grey),
                        filled: true,
                        fillColor: Colors.white,
                        border: OutlineInputBorder(
                          borderRadius: BorderRadius.circular(10),
                        ),
                      ),
                    ),
                  ],
                ),
              ),

              // Console Section (Arrow Button)
              Padding(
                padding: const EdgeInsets.only(bottom: 20),
                child: Align(
                  alignment: Alignment.bottomRight,
                  child:
                      context.watch<UserProvider>().name.trim().isNotEmpty
                          ? SizedBox(
                            width: 50,
                            height: 50,
                            child: IconButton(
                              onPressed: onPress,
                              icon: Icon(Icons.arrow_forward), // Arrow icon
                              iconSize: 30,
                              color: CustomColors.dark,
                              style: ButtonStyle(
                                backgroundColor: WidgetStatePropertyAll<Color>(
                                  Colors.white,
                                ),
                              ),
                            ),
                          )
                          : Container(),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
